test/util.dart

```diff
@@ -175,6 +177,7 @@ class RefTestCaseCollection<T> {
         createJsRefObjectCase(),
       ];
 
+  // ignore: avoid_shadowing_type_parameters
   RefTestCase createCaseByName<T>(String name) => createAllCases().singleWhere((c) => c.name == name);
```

Comment:
This actually seems like a bad shadowed type parameter that shouldn't be ignored; I think it should removed
```suggestion
  RefTestCase createCaseByName(String name) => createAllCases().singleWhere((c) => c.name == name);
```